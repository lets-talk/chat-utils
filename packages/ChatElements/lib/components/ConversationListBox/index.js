import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Header from '../Header';
import ConversationList from '../ConversationList/Loadable';
import Button from '../Button';
import './index.scss';

const ConversationListBox = (props) => {
  const {
    conversations,
    title,
    subtitle,
    avatar,
    className,
    toggleChat,
    newConversationHandler,
    clickConversationHandler,
    openMenu,
    showMinimizeButton,
    showMenuButton,
    noMoreDataText,
    emptyStateText,
  } = props;

  return (
    <div className={classNames('conversation-listbox-container', className)}>
      <Header
        title={title}
        subtitle={subtitle}
        avatar={avatar}
        toggleChat={toggleChat}
        openMenu={openMenu}
        showMinimizeButton={showMinimizeButton}
        showMenuButton={showMenuButton}
      />
      <ConversationList
        conversations={conversations}
        clickItem={clickConversationHandler}
        noMoreDataText={noMoreDataText}
        emptyStateText={emptyStateText}
        toBottomHeight={props.toBottomHeight}
      />
      <Button
        type="submit"
        clickHandler={newConversationHandler}
        value="Nueva conversaciÃ³n"
      />
    </div>
  );
};

ConversationListBox.propTypes = {
  conversations: PropTypes.array.isRequired,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  avatar: PropTypes.string,
  className: PropTypes.string,
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  clickConversationHandler: PropTypes.func,
  newConversationHandler: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
  noMoreDataText: PropTypes.bool,
  emptyStateText: PropTypes.bool,
};

ConversationListBox.defaultProps = {
  title: '',
  subtitle: '',
  avatar: '',
  toggleChat: null,
  openMenu: null,
  clickConversationHandler: null,
  newConversationHandler: null,
  showMinimizeButton: false,
  showMenuButton: false,
};

export default ConversationListBox;
