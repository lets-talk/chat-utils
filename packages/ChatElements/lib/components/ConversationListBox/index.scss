@import "../../common/styles/variables";
@import './config.scss';

.conversation-listbox-container * {
  font-family: $preferred-font;
}

.conversation-listbox-container {
  @include conversation-listbox-container-fs;
  @include animation(0, 0.5s, slide-in);
}
