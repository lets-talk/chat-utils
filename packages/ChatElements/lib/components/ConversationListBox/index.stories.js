import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';

import { wrapWithThemes } from '../../utils/stories';
// Component to show on storybook
import { ConversationListBox } from '../../../lib';

const stories = storiesOf('ConversationList', module);
stories.addDecorator(withKnobs);

const conversationList = [
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      content: 'Lorem ipusum la la ejemplo largo',
    },
    tags: [
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      type: 'important',
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo',
    },
    tags: null,
  },
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      readed: false,
      content: 'Lorem ipusum la la ejemplo largo',
    },
    tags: [
      {
        name: 'SPU-PYME',
      },
    ],
  },
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo',
    },
    tags: [
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      type: 'internal',
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo',
    },
    tags: [
      {
        name: 'SPU-PYME',
      },
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      content: 'Lorem ipusum la la ejemplo largo',
    },
  },
];

// This HOC Wraps the Box so it is inside a ThemeProvider for styled-components
// and also wraps it inside a div with the selected global themes
const ConversationListBoxWithTheme = wrapWithThemes(ConversationListBox);

stories.addWithInfo(
  'Simple Conversation List',
  'This is the basic usage of a ConversationList with providing a label to show the text.',
  () => (
    <ConversationListBoxWithTheme
      themeName={select('Theme', { light: 'LightTheme', dark: 'DarkTheme' }, 'light')}
      title={text('Title', 'Conversaciones')}
      subtitle={text('Subtitle', '')}
      senderPlaceHolder={text('Sender PlaceHolder', 'Escribe una respuesta')}
      toggleChat={action('toggleChat')}
      openMenu={action('openMenu')}
      showMenuButton={boolean('Show Menu Button', false)}
      showMinimizeButton={boolean('Show Minimize Button', false)}
      conversations={conversationList}
    />
  )
);
