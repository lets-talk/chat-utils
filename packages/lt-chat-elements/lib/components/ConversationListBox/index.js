import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Header from '../Header';
import ConversationList from '../ConversationList/Loadable';
import Button from '../Button';

const ConversationListBoxContainer = styled.div`
  border-radius: 10px 10px 0px 0px;
  box-shadow: 0px 0px 8px 1px rgba(0,0,0,0.44);
  height: 100%;
  width: 100%;
  margin: 0;
  max-width: none;
  padding-bottom: 10px;
`;

const ConversationListBox = (props) => {
  const {
    conversations,
    title,
    subtitle,
    person,
    className,
    toggleChat,
    newConversationHandler,
    clickConversationHandler,
    openMenu,
    showMinimizeButton,
    showMenuButton,
    showAddConversationButton,
    noMoreDataText,
    emptyStateText,
    newConversationText,
  } = props;

  return (
    <ConversationListBoxContainer className={className}>
      <Header
        title={title}
        subtitle={subtitle}
        person={person}
        toggleChat={toggleChat}
        openMenu={openMenu}
        showMinimizeButton={showMinimizeButton}
        showMenuButton={showMenuButton}
      />
      <ConversationList
        conversations={conversations}
        clickItem={clickConversationHandler}
        noMoreDataText={noMoreDataText}
        emptyStateText={emptyStateText}
      />
      {showAddConversationButton &&
        <Button
          type="submit"
          clickHandler={newConversationHandler}
          value={newConversationText}
        />
      }
    </ConversationListBoxContainer>
  );
};

ConversationListBox.propTypes = {
  /**
   * conversations: Conversation list to be displayed
   */
  conversations: PropTypes.array.isRequired,
  /**
   * title: Title to show on the header
   */
  title: PropTypes.string,
  /**
   * subtitle: Title to show on the header
   */
  subtitle: PropTypes.string,
  /**
   * persona: Avatar image to display on the header of the component
   */
  person: PropTypes.object,
  /**
   * toggleChat: Callback function called when user clicks on the toggle button
   */
  toggleChat: PropTypes.func,
  /**
   * openMenu: Callback function called when user clicks on open menu button
   */
  openMenu: PropTypes.func,
  /**
   * clickConversationHandler: Callback function called when user clicks on a conversation item
   * This callback recieves the conversation object, the index, and the pure dom event
   * So it should be of the form callback = (conversation, index, e) =>
   */
  clickConversationHandler: PropTypes.func,
  /**
   * clickConversationHandler: Callback function called when user clicks on a new conversation button
   * This only applies when showAddConversationButton is true
   */
  newConversationHandler: PropTypes.func,
  /**
   * showMinimizeButton: Boolean indicates wheter or not to display the minimize button on the header
   */
  showMinimizeButton: PropTypes.bool,
  /**
   * showMenuButton: Boolean indicates wheter or not to display the menu button on the header
   */
  showMenuButton: PropTypes.bool,
  /**
   * showMenuButton: Boolean indicates wheter or not to display the add conversation button
   */
  showAddConversationButton: PropTypes.bool,
  /**
   * noMoreDataText: Text to be displayed when there is no more conversation items (at the end of the conversation list)
   */
  noMoreDataText: PropTypes.string,
  /**
   * newConversationText: Text to be displayed inside the Add Conversation Button
   */
  newConversationText: PropTypes.string,
  /**
   * emptyStateText: Text to be displayed when there are no conversations to display
   */
  emptyStateText: PropTypes.string,
  /**
   * className: Extra className to provide to the component in order to style it when used in different contexts.
   */
  className: PropTypes.string,
};

ConversationListBox.defaultProps = {
  title: '',
  subtitle: '',
  person: {},
  toggleChat: null,
  openMenu: null,
  clickConversationHandler: null,
  newConversationHandler: null,
  showMinimizeButton: false,
  showMenuButton: false,
  showAddConversationButton: true,
  noMoreDataText: '',
  newConversationText: '',
  emptyStateText: '',
  className: 'LT-ConversationListBox-Wrapper',
};

export default ConversationListBox;
