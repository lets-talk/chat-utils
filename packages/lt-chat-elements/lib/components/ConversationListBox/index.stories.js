import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select, object } from '@storybook/addon-knobs';

import { WrapWithTheme } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';
// Component to show on storybook
import { ConversationListBox } from '../../../lib';

const stories = storiesOf('ConversationList', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ConversationListBox'));

const conversationList = [
  {
    id: 1,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-20T09:43:57.000-0300',
    },
    tags: [
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    id: 2,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      type: 'important',
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo, este es muy largo deberia bloquear el tamaño del mensaje la UI.',
      created_at: '2017-11-20T09:40:57.000-0300',
    },
    tags: null,
  },
  {
    id: 3,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      readed: false,
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-20T09:33:57.000-0300',
    },
    tags: [
      {
        name: 'SPU-PYME',
      },
    ],
  },
  {
    id: 4,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-19T09:43:57.000-0300',
    },
    tags: [
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    id: 5,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      type: 'internal',
      readed: true,
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-19T09:41:57.000-0300',
    },
    tags: [
      {
        name: 'SPU-PYME',
      },
      {
        name: 'SPU-CLAVE',
      },
    ],
  },
  {
    id: 6,
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-19T09:42:50.000-0300',
    },
  },
];

stories.addWithInfo(
  'Simple Conversation List',
  'This is the basic usage of a ConversationList with providing a label to show the text.',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: '400px',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', { default: 'DefaultTheme', light: 'LightTheme', dark: 'DarkTheme' }, 'default')}
      >
        <ConversationListBox
          title={text('title', 'Conversaciones')}
          subtitle={text('subtitle', '')}
          toggleChat={action('toggleChat')}
          openMenu={action('openMenu')}
          clickConversationHandler={action('clickConversationHandler')}
          newConversationHandler={action('newConversationHandler')}
          showMenuButton={boolean('showMenuButton', false)}
          showMinimizeButton={boolean('showMinimizeButton', false)}
          showAddConversationButton={boolean('showAddConversationButton', true)}
          emptyStateText={text('emptyStateText', 'Sin conversaciones abiertas')}
          avatar={text('avatar', '')}
          avatarStatus={text('avatarStatus', '')}
          noMoreDataText={text('noMoreDataText', 'No hay mas conversaciones')}
          newConversationText={text('newConversationText', 'Nueva conversación')}
          conversations={object('conversations', conversationList)}
          toBottomHeight="100%"
        />
      </WrapWithTheme>
    </div>
  )
);
