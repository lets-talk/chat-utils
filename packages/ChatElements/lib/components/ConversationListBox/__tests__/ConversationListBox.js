import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConversationListBox from '../index';

describe('ConversationListBox component', () => {
  const conversations = [
    {
      client: {
        name: 'Georgia Harmon',
        avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
      },
      last_message: {
        content: 'Lorem ipusum la la ejemplo largo',
        created_at: '2017-11-20T09:43:57.000-0300',
      },
      tags: [
        {
          name: 'SPU-CLAVE',
        },
      ],
    },
  ];

  it('should render with minimal props', () => {
    const component = shallow(<ConversationListBox
      conversations={conversations}
      newConversationHandler={() => null}
    />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      conversations,
      title: 'Tus conversaciones',
      subtitle: '',
      avatar: '',
      toggleChat: null,
      openMenu: null,
      clickConversationHandler: () => null,
      newConversationHandler: () => null,
      showMinimizeButton: false,
      showMenuButton: false,
    };

    const component = shallow(<ConversationListBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
