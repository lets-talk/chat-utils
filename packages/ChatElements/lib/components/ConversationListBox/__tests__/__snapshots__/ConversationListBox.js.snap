// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationListBox component should render with custom props 1`] = `
<div
  className="conversation-listbox-container"
>
  <Header
    avatar=""
    openMenu={null}
    showMenuButton={false}
    showMinimizeButton={false}
    subtitle=""
    title="Tus conversaciones"
    toggleChat={null}
  />
  <ConversationList
    clickItem={[Function]}
    conversations={
      Array [
        Object {
          "client": Object {
            "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
            "name": "Georgia Harmon",
          },
          "last_message": Object {
            "content": "Lorem ipusum la la ejemplo largo",
            "created_at": "2017-11-20T09:43:57.000-0300",
          },
          "tags": Array [
            Object {
              "name": "SPU-CLAVE",
            },
          ],
        },
      ]
    }
  />
  <Button
    clickHandler={[Function]}
    type="submit"
    value="Nueva conversación"
  />
</div>
`;

exports[`ConversationListBox component should render with minimal props 1`] = `
<div
  className="conversation-listbox-container"
>
  <Header
    avatar=""
    openMenu={null}
    showMenuButton={false}
    showMinimizeButton={false}
    subtitle=""
    title=""
    toggleChat={null}
  />
  <ConversationList
    clickItem={null}
    conversations={
      Array [
        Object {
          "client": Object {
            "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
            "name": "Georgia Harmon",
          },
          "last_message": Object {
            "content": "Lorem ipusum la la ejemplo largo",
            "created_at": "2017-11-20T09:43:57.000-0300",
          },
          "tags": Array [
            Object {
              "name": "SPU-CLAVE",
            },
          ],
        },
      ]
    }
  />
  <Button
    clickHandler={[Function]}
    type="submit"
    value="Nueva conversación"
  />
</div>
`;
